src/Sylius/Bundle/ApiBundle/Provider/CompositePaymentConfigurationProvider.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Provider;
+
+use Sylius\Bundle\ApiBundle\Payment\ApiPaymentMethodHandlerInterface;
+use Sylius\Component\Core\Model\PaymentInterface;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+
+/** @experimental */
+final class CompositePaymentConfigurationProvider implements PaymentConfigurationProviderInterface
+{
+    /** @var iterable<ApiPaymentMethodHandlerInterface> */
+    private $apiPaymentMethodHandlers;
+
+    public function __construct(iterable $apiPaymentMethodHandlers)
+    {
+        $this->apiPaymentMethodHandlers = $apiPaymentMethodHandlers;
+    }
+
+    public function provide(PaymentInterface $payment): array
+    {
+        /** @var PaymentMethodInterface $paymentMethod */
+        $paymentMethod = $payment->getMethod();
+
+        foreach ($this->apiPaymentMethodHandlers as $apiPaymentMethodHandler) {
+            if ($apiPaymentMethodHandler->supports($paymentMethod)) {
```

Comment:
Maybe instead of the support method, we can trigger UnsuppportedPaymentMethodException(a custom one) or use something more like delegator patter. The second approach would mean, that we are injecting an array of services but each item is identified by some key (like payment method config). In both cases, we can drop `supports` method and composite itself can implement the same interface as end providers itself.

Reply:
i would say it all defends itslef, we have handlers. If they support payment method it means they can provide data based on payment method. its clearer to read if supports then do something rather that inserting it into try catch block.